ProfileManager.js

```diff
@@ -115,6 +125,82 @@ export default class ProfileManager {
     return await this._getAgentContent({profileAgentRecord});
   }
 
+  /**
+   * Gets the profile agent assigned to the account associated with the
+   * authenticated session for the profile identified by the given profile ID.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} options.profileId - The ID of the profile to get the
+   *   profile agent for.
+   * @param {string} options.capabilityInvoker - The type of invoker to sign
+   * capability invocations.
+   *
+   * @returns {Promise<object>} The profile agent.
+   */
+  async getCapability(
+    {id, profileAgent, profileId, capabilityInvoker = 'local'} = {}) {
+    const capabilityCacheKey = `profileAgent-${profileAgent.id}-zcaps`;
+    const capabilityKey = `${capabilityCacheKey}-${id}-${capabilityInvoker}`;
+
+    if(!this._cache.has(capabilityCacheKey)) {
+      this._cache.set(capabilityCacheKey, new Cache());
+    }
+
+    let promise = this._requests.get(capabilityKey);
```

Comment:
It seems like we're missing one or more encapsulating concepts here such as a `RequestQueue` ... and perhaps one that combined a RequestQueue with a cache which would mean we wouldn't need the `capabilityKey` to encode which cache it was a part of (like a `CapabilityCache` or something). That could be cleaner both to understand and to maintain. I imagine using its API would also clean up this function considerably.

Reply:
I could create a `RequestQueue`, but what I really want is a decorator for all the functions that need that functionality. Does your suggestion for a `CapabilityCache` mean you want its existence to be independent of the cache for the ProfileManager? I intentionally made it so that the `capabilityCache` was apart of the ProfileManager's global cache. This was done to ensure that when the cache for the ProfileManager was cleared, so was everything else.

Reply:
It can still be cleared ... but the way the global cache is setup now is also weird. It's like we're clearing entirely different sets of things. Like, if you use the edv client cache, we'll keep it around, but if don't, we'll wipe the whole thing? The top level cache being an LRU cache is really confusing to me. It probably should be a map of caches that gets cleared all together when the session changes.

Reply:
Replaced the erroneous top level cache with a Map.